operator/pkg/component/component/component.go

```diff
@@ -728,7 +705,43 @@ func createHelmRenderer(c *CommonComponentFields) (helm.TemplateRenderer, error)
 	return helm.NewHelmRenderer(iop.InstallPackagePath, helmSubdir, cns, c.Namespace)
 }
 
+// isComponentEnabled reports whether the component is enabled.
+func isComponentEnabled(c *CommonComponentFields) (bool, error) {
+	var err error
+	enabled := false
+	switch {
+	case c.componentName.IsCoreComponent():
+		enabled, err = c.Translator.IsComponentEnabled(c.componentName, c.InstallSpec)
+		if err != nil {
+			return false, err
+		}
+
+	case c.componentName.IsGateway():
+		enabled = boolValue(c.gatewaySpec.Enabled)
```

Comment:
I don't understand this. Why only read from the spec? Why not handle the same way as other components and look at both spec and values?

Reply:
good question! we had to push the logic for enablement up to the code where user overlay is applied. the reason is a problem of precedence - we only want the values enablement to override IOS if it came from the user. to simplify things, we translate that value into IOP at that time and use IOP fields only from that point. 
it's really not a great solution so if you can figure out something better here i'd love to improve this.